### PR TITLE
Reintroduce operator precedence fix with legacy override

### LINKED ISSUE #893: Unary minus binds too loosely
Consider the following expression:

```
{{ -1 is integer }}
```

I would expect this to evaluate to `true`, since `-1` is an `integer`. However, this actually throws an error.

```
Exception in thread "main" com.hubspot.jinjava.interpret.FatalTemplateErrorsException: Cannot negate 'class java.lang.Boolean'
	at com.hubspot.jinjava.Jinjava.render(Jinjava.java:192)
	at Scratch.main(scratch.java:6)
```

This implies that the unary minus is binding looser than the `is` operator. That is, Jinjava interprets that expression the same as `-(1 is integer)` when it should interpret it as `(-1) is integer`.

### PR DESCRIPTION
**Pull request by tkindy**
Closes #893 

This PR is basically https://github.com/HubSpot/jinjava/pull/895 but with the changes disabled by default. They can be enabled by setting a new flag on `LegacyOverrides` as part of the `JinjavaConfig`.

### CODE REVIEW

**Comment by tkindy** [comment]
Actually, I want to move some of the tests around first.

File: src/main/java/com/hubspot/jinjava/el/ext/ExtendedParser.java [hunk 1]
```diff
@@ -578,4 +628,20 @@ public AstNode createAstNode(AstNode... children) {
...
+        .getConfig()
+        .getLegacyOverrides()
+        .isUseNaturalOperatorPrecedence()
+    );
+  }
+
+  @FunctionalInterface
+  private interface ParseLevel {
+    AstNode apply(boolean required) throws ScanException, ParseException;
+  }
```
Comment (jasmith-hs) [inline src/main/java/com/hubspot/jinjava/el/ext/ExtendedParser.java:646]: I like this!

File: src/main/java/com/hubspot/jinjava/el/ext/ExtendedParser.java [hunk 2]
```diff
@@ -578,4 +628,20 @@ public AstNode createAstNode(AstNode... children) {
...
+
+  private static boolean shouldUseNaturalOperatorPrecedence() {
+    return (
+      JinjavaInterpreter.getCurrent() != null &&
+      JinjavaInterpreter
+        .getCurrent()
+        .getConfig()
+        .getLegacyOverrides()
+        .isUseNaturalOperatorPrecedence()
+    );
```
Comment (jasmith-hs) [inline src/main/java/com/hubspot/jinjava/el/ext/ExtendedParser.java:640]: Theres
`JinjavaInterpreter.getCurrentMaybe().map(interpreter -> interpreter.getConfig().getLegacyOverrides().isUseNaturalOperatorPrecedence()).orElse(false)` too if you're big on writing expressions like that

**Review by jasmith-hs — APPROVED** [review]

File: src/main/java/com/hubspot/jinjava/el/ext/ExtendedParser.java (hunk 2 above)
Comment (tkindy) [reply src/main/java/com/hubspot/jinjava/el/ext/ExtendedParser.java:640]: Ooh I very much am, thanks! 😄 I don't know why I didn't think to just to an `Optional.ofNullable` myself.

File: src/main/java/com/hubspot/jinjava/el/ext/ExtendedParser.java (hunk 1 above)
Comment (tkindy) [reply src/main/java/com/hubspot/jinjava/el/ext/ExtendedParser.java:646]: Thanks, I'm glad! 😄

**Pull request merged**